source/common/unified_malloc_framework/src/memory_provider.cpp

```diff
@@ -46,6 +54,51 @@ umfMemoryProviderCreate(struct umf_memory_provider_ops_t *ops, void *params,
     return UMF_RESULT_SUCCESS;
 }
 
+enum umf_result_t umfMemoryProviderRegister(umf_memory_provider_ops_t *ops) {
+
+    // TODO check if this provider isn't already registered
+    globalProviders.push_back(*ops);
+
+    return UMF_RESULT_SUCCESS;
+}
+
+enum umf_result_t umfMemoryProvidersRegisterGetNames(char *providers,
```

Comment:
I don't think names are important. I'd create some opaque 'factory' type that'd return here in a simple array. This factory type would then have methods like `...get_name` or `...get_ops`.
Alternatively, you could move the selection of providers based on name or some other *abstracted* config here.

Returning everything in a string array just to parse it higher up the stack looks like an unnecessary complication.